hudi-client/src/main/java/org/apache/hudi/client/HoodieReadClient.java

```diff
@@ -144,6 +163,43 @@ private void assertSqlContext() {
     }
   }
 
+  private synchronized void startEmbeddedServerView() {
```

Comment:
if we have to copy this code, then breaking the abstraction does not feel right?

I actually think its okay for ReadClient to not use the TimelineServer..

Reply:
My original thought, the first step is to decouple the inheritance chain. I did not change any logic so that I would not touch anything about the key things. Then I will try to remove some unnecessary logic.

Reply:
I found keep timeline server specific logic, we need to change the code in another module to catch the `exception`. So, all right, I have removed the timeline server in read client.

Reply:
The Travis is green, please review again.